servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/gc/AbstractRestGCRepoTest.java

```diff
@@ -0,0 +1,192 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.RowFactory;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.catalyst.analysis.NoSuchTableException;
+import org.junit.jupiter.api.Test;
+import org.projectnessie.client.api.NessieApiV1;
+import org.projectnessie.error.NessieConflictException;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.gc.base.IdentifiedResult;
+import org.projectnessie.gc.base.IdentifiedResultsRepo;
+import org.projectnessie.model.Branch;
+import org.projectnessie.model.IcebergTable;
+
+public abstract class AbstractRestGCRepoTest extends AbstractRestGC {
+
+  private final String catalogName = "nessie";
+  private final String identifierNameSpace = "db1";
+  private final String identifierTableName = "identified_results";
+  private final String identifier = identifierNameSpace + "." + identifierTableName;
+  private final String refName = "someRef";
+  private final String gcRefName = "someGcRef";
+  private final String catalogAndIdentifierWithReference = getCatalogAndIdentifierWithReference();
+
+  @Test
+  public void testGCRepoInProgressAndSuccess() {
+    checkAndCreateGcReference(getApi());
+    try (SparkSession sparkSession = getSparkSession()) {
+      IdentifiedResultsRepo identifiedResultsRepo =
+          new IdentifiedResultsRepo(sparkSession, catalogName, gcRefName, identifier);
+      String runId = UUID.randomUUID().toString();
+      Timestamp startAt = Timestamp.from(Instant.now());
+      // write 5 rows to table without a marker row
+      writeRows(sparkSession, identifiedResultsRepo, getRows(runId, startAt, 5));
+
+      // try to collect in-progress results.
+      IdentifiedResult identifiedResult = identifiedResultsRepo.collectExpiredContents(runId);
+      // should not collect any results as write is in-progress.
+      assertThat(identifiedResult.getContentValues()).isEmpty();
+
+      // latest completed run id should not be present.
+      assertThat(identifiedResultsRepo.getLatestCompletedRunID()).isNull();
+
+      // write marker row
+      writeMarkerRow(sparkSession, identifiedResultsRepo, runId);
+
+      assertThat(identifiedResultsRepo.getLatestCompletedRunID()).isEqualTo(runId);
+
+      // collect results as marker row is written.
+      identifiedResult = identifiedResultsRepo.collectExpiredContents(runId);
+      // should collect 5 rows.
+      assertThat(identifiedResult.getContentValuesForReference(refName).size()).isEqualTo(5);
+
+      sparkSession.sql(String.format("DROP TABLE %s", catalogAndIdentifierWithReference));
```

Comment:
we should use before/after or move it to finally. It maybe not dropped in case there will be an exception before (same for other tests)

Reply:
As it was a test code and test was expected to pass, I have added like that.

Moved to finally block now.